etc/gerbil.el

```diff
@@ -203,15 +198,16 @@
         (with-current-buffer buf
           (goto-char (point-max))
           (when (re-search-backward gerbil-compile-mark-rx nil t)
-            (let (limit (point))
+            (let ((limit (point)))
               (goto-char (point-max))
               (when (re-search-backward gerbil-error-locat-rx limit t)
                 (let* ((loc (gerbil-extract-locat (buffer-substring (point) (point-max))))
                        (fname (if gerbil-build-directory
                                   (concat gerbil-build-directory (car loc))
                                 (car loc))))
                   (find-file fname)
-                  (goto-line (cadr loc))
+                  (goto-char (point-min))
+                  (forward-line (1- (cadr loc)))
```

Comment:
what was wrong with `goto-line`? Just curious, i am fine with changing it as long as we are not breaking anything.

Reply:
Right, I forgot to add a comment about this.

Basically `goto-line` is an interactive function, and the use of interactive functions is discouraged in non-interactive functions. Emacs produces a warning with this exact message when byte-compiling.

Interactive functions are generally reserved for the user to call them directly, ala `M-x` or keybindings, because they may alter the user environment (like markers, etc) but you can still use them with no warnings in non-interactive functions if you do it like this:

```elisp
(call-interactively 'goto-line nil (cadr loc))  ;; nil prevents goto-line from appearing in the user's recently used commands
```

This said, `goto-line`'s function [documentation](https://github.com/emacs-mirror/emacs/blob/master/lisp/simple.el#L1229) recommends what I used as a better alternative for this use case:
```
[...]
This function is usually the wrong thing to use in a Lisp program.
What you probably want instead is something like:
  (goto-char (point-min))
  (forward-line (1- N))
[...]
```
So in this case I am still trusting the minds behind `goto-line` 😄 (and in my tests nothing seems to be broken)

Reply:
thanks, makes sense now.